src/Components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import { Todo } from '../../react-app-env';
+
+import users from '../../api/users';
+
+interface Props {
+  addTodo: (todo: Todo) => void;
+}
+
+interface State {
+  inputTitle: string,
+  selectedName: string,
+  isTitle: boolean,
+  isName: boolean,
+}
+
+export class TodoForm extends React.Component<Props, State> {
+  state: State = {
+    inputTitle: '',
+    selectedName: '',
+    isTitle: false,
+    isName: false,
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>) => {
+    const { name, value } = event.target;
+
+    if (name === 'inputTitle') {
+      this.setState({
+        [name]: value,
+      });
+    }
+
+    if (name === 'selectedName') {
+      this.setState({
+        [name]: value,
+      });
+    }
+  };
+
+  clearState = () => {
+    this.setState({
+      inputTitle: '',
+      selectedName: '',
+      isTitle: false,
+      isName: false,
+    });
+  };
+
+  submitTodo = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    const {
+      inputTitle,
+      selectedName,
+    } = this.state;
+
+    if (!inputTitle) {
+      this.setState({ isTitle: true });
+    }
+
+    if (!selectedName) {
+      this.setState({ isName: true });
+    }
+
+    if (inputTitle && selectedName) {
+      const newTodo = {
+        title: inputTitle,
+        user: users.find(user => user.name === selectedName),
+      } as Todo;
```

Comment:
```suggestion
      const newTodo: Todo = {
        title: inputTitle,
        user: users.find(user => user.name === selectedName),
        id: uuid(),
      };
```
будет ошибка, нужно заполнить все поля из Todo